friktion/friktion/bid_details.py

```diff
@@ -1,38 +1,17 @@
+from dataclasses import dataclass
+
 from anchorpy import Wallet
 from nacl import signing  # type: ignore
 from solana.publickey import PublicKey
 
-from friktion.pda import SwapOrderAddresses
-
 
+@dataclass
 class BidDetails:
-
-    swap_order_owner: PublicKey
     order_id: int
-
     counterparty_give_pool: PublicKey
     counterparty_receive_pool: PublicKey
-
     bid_size: int
-
-    def __init__(
-        self,
-        swap_order_owner: PublicKey,
-        order_id: int,
-        counterparty_give_pool: PublicKey,
-        counterparty_receive_pool: PublicKey,
-        bid_size: int,
-        bid_price: int,
-    ):
-        self.swap_order_owner = swap_order_owner
-        self.order_id = order_id
-        self.counterparty_give_pool = counterparty_give_pool
-        self.counterparty_receive_pool = counterparty_receive_pool
-        self.bid_size = bid_size
-        self.bid_price = bid_price
-
-    def get_swap_order_address(self) -> PublicKey:
-        return (SwapOrderAddresses(self.swap_order_owner, self.order_id)).swap_order_address
+    bid_price: int
 
     def as_signed_msg(
         self, wallet: Wallet, receive_amount: int, give_amount: int
```

Comment:
should we add bid_price to the signature?

Reply:
give_amount is the # of tokens the swap order creator is giving, receive_amount is # of tokens the swap order creator is receiving. So, receive token === bid token

Reply:
Nice, fixed that.

1) Will `receive_amount == give_amount` ?

2) should we add `bid_price` to be signed?

Reply:
1. no! they will almost never be equal, since bid amount is quite different from offer amount. To be clear `give amount` == `offer amount` and `receive_amount` == `bid_amount`. We can rename these if easier?

Reply:
Yeah, renaming it may help to understand.
Sorry, I am still a little confused. 😅 so `give_amount = self.bid_size` and `receive_amount = self.bid_price * self.bid_size`?

Reply:
Hi @bigdefidev - just to help here clearing up the confusion.

We currently have "buyAmount" and "sellAmount".

- buyAmount = quantity of tokens offered, which should match your `give_amount`
- sellAmount = buyAmount * price

One thing I don't understand is why your `receive_amount` might differ from the `give_amount`?

Reply:
receive_amount = give_amount * price = sellAmount != give_amount.

Does that make sense?

Reply:
"give_amount = self.bid_size and receive_amount = self.bid_price * self.bid_size" @cassios this should be true, but I think the way I wrote things is confusing. I'll rename them for good measure.

Reply:
Nice. So we can adjust `as_signed_msg` to use the parameters from the object. I will push a commit for that